### PR TITLE
[postgres-connector] Add proper build and testing around package

### PR DESCRIPTION
**Pull request by cf19drofxots**
This ensures the postgres package can be built and consumed as expected, as well as introducing tests to verify expected functionality by the implementation.

This also achieves a couple other things:
- uses in-memory store and messaging services for the Node tests instead of Firebase
  - follow up PRs can fully eliminate the need for Firebase across the stack
- exposes another method on the Node for retrieving a proposed `AppInstance` (primarily for easier testing purposes)

### CODE REVIEW

**Comment by cf19drofxots** [comment]
To confirm that the postgres implementation actually works, this PR adds a test to the node package consuming the postgres package.

This is currently failing because the relational APIs don't offer value lookup based on partial path keys.

File: .circleci/config.yml [hunk 1]
```diff
@@ -7,6 +7,15 @@ defaults: &defaults
   working_directory: /home/circleci/project
   docker:
     - image: circleci/node:10.15.3
+      environment:
```
Comment (snario) [inline .circleci/config.yml]: Why are you adding all of these to the `defaults`?

File: packages/node/.env.defaults [hunk 2]
```diff
@@ -4,3 +4,9 @@ FIREBASE_MESSAGING_SERVER_KEY="messageKey"
 FIREBASE_STORE_SERVER_KEY="storeKey"
 FIREBASE_STORE_PREFIX_KEY="dev-store"
 GANACHE_PORT="8545"
+POSTGRES_USER="postgres"
```
Comment (snario) [inline packages/node/.env.defaults]: It feels a bit odd that we have default environment variables for the `node` package that have to do with plugins being used by it (presumably in tests?)

File: packages/node/package.json [hunk 3]
```diff
@@ -16,12 +16,12 @@
   "devDependencies": {
     "@counterfactual/apps": "0.1.1",
     "@counterfactual/firebase-server": "0.0.1",
+    "@counterfactual/postgresql-node-connector": "0.0.2",
```
Comment (snario) [inline packages/node/package.json]: Seems odd that the Node has a dev dependency on a plugin to it.

Maybe the plugin should have a dev dependency on the node for testing purposes instead.

File: packages/node/test/integration/get-app-instance.spec.ts [hunk 4]
```diff
@@ -4,7 +4,7 @@ import { Node as NodeTypes } from "@counterfactual/types";
 
 import { Node } from "../../src";
 
-import { setup } from "./setup";
+import { setupWithFirebaseServiceFactory } from "./setup";
```
Comment (snario) [inline packages/node/test/integration/get-app-instance.spec.ts]: On second thought, why are we using Firebase for local Node testing? Why can't we just use in-memory messaging and storage?

File: packages/node/test/integration/setup.ts [hunk 5]
```diff
@@ -14,7 +16,33 @@ import { MNEMONIC_PATH, Node } from "../../src";
 import { CF_PATH } from "../global-setup.jest";
 import { A_MNEMONIC, B_MNEMONIC } from "../test-constants.jest";
 
-export async function setup(
+export async function setupWithFirebaseServiceFactory(
```
Comment (snario) [inline packages/node/test/integration/setup.ts]: In general the concept of "service factories" feels like it shouldn't exist. It just _happens_ to be the case that Firebase can be used in a hacky way to do messaging, but in general messaging and storage are fundamentally different things and trying to introduce this kind of "service factory" wrapper concept feels like way too much overhead for this weird edge case.

File: packages/node/test/integration/setup.ts [hunk 6]
```diff
@@ -26,6 +54,37 @@ export async function setup(
   const messagingService = firebaseServiceFactory.createMessagingService(
     process.env.FIREBASE_MESSAGING_SERVER_KEY!
   );
+
+  const postgresServiceFactory = new PostgresServiceFactory({
```
Comment (snario) [inline packages/node/test/integration/setup.ts]: As evidenced here. Why do we have a "PostgresServiceFactory"? We just need a "PostgresStoreService".

File: .circleci/config.yml (hunk 1 above)
Comment (cf19drofxots) [reply .circleci/config.yml]: Just to get it working. This is still in draft.

File: packages/node/package.json (hunk 3 above)
Comment (cf19drofxots) [reply packages/node/package.json]: That was my initial thought and is probably the better way to do it. Only reason I opted for this in the immediate future is because the main objective was to confirm the postgres implementation works as expected, but it's also because setting up proper e2e tests that use in the Node in the postgres package requires a lot of context

File: packages/node/test/integration/get-app-instance.spec.ts (hunk 4 above)
Comment (cf19drofxots) [reply packages/node/test/integration/get-app-instance.spec.ts]: Yeah we could. I'll remove this.

File: packages/node/test/integration/setup.ts (hunk 6 above)
Comment (rhlsthrm) [reply packages/node/test/integration/setup.ts]: This was my fault, I tried to abide by the given interfaces as much as possible, the factory did seem unnecessary to me then as well. I will remove this.

Comment (rhlsthrm) [reply packages/node/test/integration/setup.ts]: Or it looks like you still might be deep into this @cf19drofxots, let me know if you want me to do any of this.

File: packages/node/test/integration/setup.ts (hunk 5 above)
Comment (cf19drofxots) [reply packages/node/test/integration/setup.ts]: The nice property of a service factory is that it enables much easier testing as we'd need multiple instances of a service (this is only true for the store service, not the messaging service).

File: packages/node/test/integration/setup.ts (hunk 6 above)
Comment (cf19drofxots) [reply packages/node/test/integration/setup.ts]: As said in the other comment, this gives more flexibility in creating services, as compared to creating multiple services separately which would amount to effectively the same thing as having a factory producing the services.

**Comment by cf19drofxots** [comment]
So this can be merged as is, though I'm currently working on separating out the ganache environment dependency from the Node package so that the proper context can be injected into the Node _in another_ package, which is the motivation for moving the postgres tests outside of the Node package and into the postgres package.

/cc @snario

File: .circleci/config.yml [hunk 7]
```diff
@@ -52,12 +52,25 @@ jobs:
             - packages/*/build
 
   run-non-playground-tests:
-    <<: *defaults
+    docker:
+      - image: circleci/node:10.15.3
```
Comment (snario) [inline .circleci/config.yml]: Use `circleci-environment` image

File: .circleci/config.yml [hunk 8]
```diff
@@ -52,12 +52,25 @@ jobs:
...
+        environment:
+          POSTGRES_HOST: 127.0.0.1
+          POSTGRES_USER: postgres
+          POSTGRES_DATABASE: postgres
+          POSTGRES_PORT: 5432
+
+      - image: circleci/postgres:9.6.2-alpine
```
Comment (snario) [inline .circleci/config.yml]: Why are there two images?

Comment (cf19drofxots) [reply .circleci/config.yml]: The second image runs in the background and provides a Postgres server.
